feathr_project/feathr/api/app/core/feathr_api_exception.py

```diff
@@ -1,8 +1,17 @@
-from app.core.error_code import ErrorCode
-
+from feathr.api.app.core.error_code import ErrorCode
```

Comment:
Looking around the code, it seems that the package root is `feathr`, not `app`, so I corrected the `from app.core.error_code import ErrorCode` while I was here.

Reply:
@YihuiGuo can comment on this. But as long as your tests passes, it should be ok to commit.

Reply:
thanks -- looking forward to what @YihuiGuo  says here, as it is a little scary to have two different root namespaces floating around in the code!

Reply:
Those code with namespace starts with app are introduced for feathr rest api which supports feathr ui. I agree have two different root namespaces is causing confusion.

Ideally, we should move api folder out of feathr. The code under api folder does not needs to be included in feathr package as rest api is meant to be deployed as a container app, instead of being consumed as a library. It's okay to feathr rest api have a different root namespace than feathr core as it is a standalone application consumes feathr core. 

I will do the code move when the prototype work for current rest api is completed. In the meanwhile, could you please make sure your code change in feathr core not depends on feathr rest api? Otherwise there will be circular dependency

Reply:
Thanks @blrchen!

> In the meanwhile, could you please make sure your code change in feathr core not depends on feathr rest api? Otherwise there will be circular dependency

This a good point. @hangfei asked me to put this new exception into the `api.app.core.error_code` module (see [PR feedback](https://github.com/lhayhurst/feathr/pull/1#discussion_r873210122)). 

I'm not sure what the right thing to do here; this PR is my first sojourn into the feathr code so I don't have good intuition into these sorts of design. As such I have no strong opinions here. Options seem to be:

1) Leave the code as is and risk having a circular dependency 
2) Back out adding the `FeatureNameValidationError` to the  `api.app.core.error_code` module ; back out exporting the exceptions from `feathr/__init__.py`, and simply define the `FeatureNameValidationError` class in `feature.py` and export the `FeatureNameValidationError` from `feathr/__init__.py`. 
3) Some other option? 

Happy to proceed however everyone sees fit! Namespace/package/module design is hard :/ 

Cheers
Lyle